src/ContentValidationListener.php

```diff
@@ -356,6 +367,101 @@ protected function useRawData($controllerService)
         return false;
     }
 
+    /**
+     * @param $controllerService
+     * @return bool
+     */
+    protected function shouldRemoveEmptyData($controllerService)
+    {
+        if (! isset($this->config[$controllerService]['remove_empty_data'])
+            || (isset($this->config[$controllerService]['remove_empty_data'])
+                && $this->config[$controllerService]['remove_empty_data'] === true)
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * @param array $data Data to filter null values from
+     * @param array $compareTo Original data, send along to preserve keys/values in $data which are intentional
+     *
+     * @return array
+     */
+    // TODO strict types for PHP > 7 when module gets bumped
+    //    protected function removeEmptyData(array $data, array $compareTo = []) : array
+    protected function removeEmptyData($data, $compareTo = [])
+    {
+        // TODO when bumped to PHP > 7 and strict types above enforced this check may be removed
+        if (! is_array($data) || ! is_array($compareTo)) {
+            throw new TypeError(
+                sprintf(
+                    'Expected array\'s for function %s. Received "%s" and "%s".',
+                    __METHOD__,
+                    gettype($data),
+                    gettype($compareTo)
+                ),
+                500,
+                __FILE__
+            );
+        }
+
+        // Callback for array_filter() to remove null values (array_filter() removes 'false' values)
+        $removeNull = function ($value, $key = null) use ($compareTo) {
+            // If comparison array is empty, do a straight comparison
+            if (empty($compareTo)) {
+                return ! is_null($value);
```

Comment:
We tend to use the construct `null !== $value`.

Reply:
done